csharp/src/Apache.Arrow/Arrays/ListArray.cs

```diff
@@ -49,17 +114,43 @@ private ListArray(ArrayData data, IArrowArray values) : base(data)
 
         public override void Accept(IArrowArrayVisitor visitor) => Accept(this, visitor);
 
+
+        [Obsolete]
         public int GetValueOffset(int index)
         {
+            if (index < 0 || index > Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
             return ValueOffsets[index];
         }
 
         public int GetValueLength(int index)
         {
+            if (index < 0 || index >= Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
             var offsets = ValueOffsets;
             return offsets[index + 1] - offsets[index];
         }
 
+        public IArrowArray GetSlicedValues(int index)
+        {
+            if (index < 0 || index >= Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+
+            //TODO: Use IArrowArray.Slice
```

Comment:
What's the plan for addressing this TODO?

Side note: In the back of my head I was considering removing some of the interfaces in this library, since they really don't add much value over the abstract base class. `IArrowArray` vs. `ArrowArray`. Thoughts?

See [https://issues.apache.org/jira/browse/ARROW-5546](https://issues.apache.org/jira/browse/ARROW-5546)

Reply:
Sorry for my ignorance of this background.
I thought [`IArrowArray.Slice`](https://github.com/Intel-bigdata/arrow/blob/master/csharp/src/Apache.Arrow/Interfaces/IArrowArray.cs#L36) was going to be implemented and we should use it after it's implemented.

I agree with removing the `IArrowArray` interface.
I'll remove this `TODO`.